lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,28 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  #use a Hash to keep track of the number of duplicates
+  new_hash = Hash.new(0)
+  #iterate through each element in the array
+  arr.each do |idx|
+      #add all elements to the new_hash and add one to the key's value if there's a duplicate
+      new_hash[idx] += 1
+  end
+  #make sure that the Array passed in contained something
```

Comment:
Nice thought here, but for the purposes of this assignment we don't have to have this check. Start small -- meet the problem specs -- and _then_ add functionality.